packages/types/src/models/V1beta1Provider.ts

```diff
@@ -0,0 +1,111 @@
+/**
+ * Forklift API
+ * Migration toolkit for virtualization (Forklift) API definitions.
+ *
+ * The version of the OpenAPI document: 2.4.0
+ * Contact: kubev2v-dev@redhat.com
+ *
+ * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
+ * https://openapi-generator.tech
+ * Do not edit the class manually.
+ */
+
+import { exists } from "../runtime";
+
+import type { V1beta1ProviderSpec } from "./V1beta1ProviderSpec";
+import {
+  V1beta1ProviderSpecFromJSON,
+  V1beta1ProviderSpecToJSON,
+} from "./V1beta1ProviderSpec";
+import type { V1beta1ProviderStatus } from "./V1beta1ProviderStatus";
+import {
+  V1beta1ProviderStatusFromJSON,
+  V1beta1ProviderStatusToJSON,
+} from "./V1beta1ProviderStatus";
+
+/**
+ *
+ * @export
+ * @interface V1beta1Provider
+ */
+export interface V1beta1Provider {
```

Comment:
Ok honest questions here...

How do we handle the differences between this generated type and the manual type https://github.com/kubev2v/forklift-console-plugin/blob/604a8878cc5f6fedf354b747797ccfec9eeca13a/packages/legacy/src/queries/types/providers.types.ts#L17-L30

Should the CRDs contain more info?

Is the intention to migrate from the legacy types to the generated types?

Reply:
> Should the CRDs contain more info?

the CRD are just what the backend guy documented in there code, it's like jdoc for go, so if they don't document a field we will not get it, the different with us manually writing interfaces is that we use the same base struct as the API (for specific vertion, in this case what will be the API for 2.4.0) so we can't invent things that are not in the API, and if the API have new fields we don't know about, we get them automatically.

> Is the intention to migrate from the legacy types to the generated types?

yes, I hope we can remove the legcy code in next version after 2.4.0

Reply:
for ref:
https://github.com/kubev2v/forklift-controller/blob/15664d6cea7e2aa2953145ff4311bdb509554f17/pkg/controller/provider/validation.go#L254

`vddkInitImage`is a key in a dictionaly, so the type is  `{ [key: string]: any }` and in the case of vmware provider the backend will look for `vddkInitImage` string as akey